lib/ace/plugin_cache.rb

```diff
@@ -75,5 +85,23 @@ def sync_core(environment)
       Puppet::Configurer::PluginHandler.new.download_plugins(env)
       libdir(File.join(environments_dir, 'plugins'))
     end
+
+    # the cache_dir will be the `cache-dir` from
+    # the ace config, with the appended environments, i.e.
+    # /opt/puppetlabs/server/data/ace-server/cache/environments
+    # then the directories within this path, which will be
+    # the puppet environments will be removed if they have
+    # not been modified in the last 7 days
+    # when the purge runs (every 24 hours)
+    def expire(purge_ttl)
+      expired_time = Time.now - purge_ttl
+      cache_dir_mutex.with_write_lock do
+        Dir.glob(File.join(cache_dir, '*')).select { |f| File.directory?(f) }.each do |dir|
+          if File.mtime(dir) < expired_time
```

Comment:
What real criteria do we want to measure on which files to remove? The current ticket design does not answer that.

Reply:
And if you're saying "modified 7 days or more ago", when do the environment directories get modified?

Reply:
For 1), what do you mean? Do you mean in terms of logging that an environment dir has been removed?

For 2), the environment directory `mtime` will be updated when that environment is used for a agentless catalog execution.

Reply:
1) I was hoping for something like "The environment hasn't been _used_ for more than 7 days"

2) oh, I wasn't aware of https://github.com/puppetlabs/ace/blob/755e51f7cd87424108b2210072e42a7577a949ce/lib/ace/plugin_cache.rb#L59